hax/minitrees.py

```diff
@@ -161,16 +218,25 @@ def get(run_name, treemaker, force_reload=False):
     if not hasattr(treemaker, '__version__'):
         raise AttributeError("Please add a __version__ attribute to treemaker %s." % treemaker_name)
     minitree_filename = "%s_%s.root" % (run_name, treemaker_name)
+    if save_pickle:
+        minitree_pickle_filename = "%s_%s.pkl" % (run_name, treemaker_name)
 
     # Do we already have this minitree? And is it good?
     minitree_path = _check_minitree_path(minitree_filename, treemaker, run_name,
                                          force_reload=force_reload)
     if minitree_path is not None:
-        return minitree_path
+        empty_frame = pd.DataFrame()
+        return minitree_path, empty_frame
 
     # We have to make the minitree file
     # This will raise FileNotFoundError if the root file is not found
     skimmed_data = treemaker().get_data(run_name)
+    
+    # Setting save_arrays to True if any arrays/vectors in DataFrame
+    for branch_name in list(skimmed_data):
+        if hasattr(skimmed_data[branch_name][0], "__len__"):
+            save_arrays = True
```

Comment:
Let's print / log a warning rather than just  turning this on silently. Maybe say something like "Branch %s is an array field: turn on save_arrays when making minitrees, or use DataExtractor.". Personally I would even throw an error rather than a warning, so the user will turn it on themselves, but that's up to you.